rs/types/types/src/crypto/threshold_sig/ni_dkg.rs

```diff
@@ -45,6 +45,19 @@ impl From<NiDkgMasterPublicKeyId> for MasterPublicKeyId {
     }
 }
 
+impl TryFrom<MasterPublicKeyId> for NiDkgMasterPublicKeyId {
+    type Error = &'static str;
+
+    fn try_from(value: MasterPublicKeyId) -> Result<Self, Self::Error> {
+        match value {
+            MasterPublicKeyId::VetKd(vet_kd_key_id) => {
+                Ok(NiDkgMasterPublicKeyId::VetKd(vet_kd_key_id))
+            }
+            _ => Err("This is not a NiDkg key"),
```

Comment:
I suggest to match explicitly on `MasterPublicKeyId::Ecdsa(_) | MasterPublicKeyId::Schnorr(_)` rather than using the catch-all so that the compiler will force us to look at this code in case we ever add another MasterPublicKeyId variant.

Reply:
Yesh, fair point, will do

Reply:
Fixed